src/MiniCover/Commands/Options/SolutionDirOption.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.Extensions.CommandLineUtils;
+using System;
+using System.IO;
+
+namespace MiniCover.Commands.Options
+{
+    public class SolutionDirOption : IMiniCoverOption<DirectoryInfo>
+    {
+        private const string DefaultValue = "./";
+        private bool _invalidated;
+        private CommandOption _option;
+        private DirectoryInfo _value;
+        public string Description => $"Solution directory [default: {DefaultValue}]";
+        public string OptionTemplate => "--solutiondir";
+        public CommandOptionType Type => CommandOptionType.SingleValue;
+
+        public DirectoryInfo Value
+        {
+            get
+            {
+                if (_invalidated) return _value;
+                throw new MemberAccessException("Option should be invalidate before Value access");
+            }
+
+            set => _value = value;
```

Comment:
This seter is not needed

Reply:
updated in #48 I can move it to this PR if necessary

Reply:
No, it's fine.
Can close this PR then and check only the other one?